pkg/adapter/kubernetes/dogucr/doguSerializer.go

```diff
@@ -0,0 +1,129 @@
+package dogucr
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/cloudogu/cesapp-lib/core"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/adapter/serializer"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/ecosystem"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domainservice"
+	v1 "github.com/cloudogu/k8s-dogu-operator/api/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func parseDoguCR(cr *v1.Dogu) (*ecosystem.DoguInstallation, error) {
+	if cr == nil {
+		return nil, &domainservice.InternalError{
+			WrappedError: nil,
+			Message:      "Cannot parse dogu CR as it is nil",
+		}
+	}
+	// parse dogu fields
+	version, versionErr := core.ParseVersion(cr.Spec.Version)
+	namespace, _, nameErr := serializer.SplitDoguName(cr.Spec.Name)
+	err := errors.Join(versionErr, nameErr)
+	if err != nil {
+		return nil, &domainservice.InternalError{
+			WrappedError: err,
+			Message:      "Cannot load dogu CR as it cannot be parsed correctly",
+		}
+	}
+	// parse persistence context
+	persistenceContext := make(map[string]interface{}, 1)
+	persistenceContext[doguInstallationRepoContextKey] = doguInstallationRepoContext{
+		resourceVersion: cr.GetResourceVersion(),
+	}
+	return &ecosystem.DoguInstallation{
+		Namespace:          namespace,
+		Name:               cr.Name,
+		Version:            version,
+		Status:             cr.Status.Status,
+		Health:             ecosystem.HealthStatus(cr.Status.Health),
+		UpgradeConfig:      ecosystem.UpgradeConfig{AllowNamespaceSwitch: cr.Spec.UpgradeConfig.AllowNamespaceSwitch},
+		PersistenceContext: persistenceContext,
+	}, nil
+}
+
+func toDoguCR(dogu *ecosystem.DoguInstallation) (*v1.Dogu, error) {
```

Comment:
🚫 **[checkstyle]** <unparam> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>toDoguCR - result 1 (error) is always nil